sound/soc/intel/common/soc-acpi-intel-icl-match.c

```diff
@@ -33,55 +33,113 @@ struct snd_soc_acpi_mach snd_soc_acpi_intel_icl_machines[] = {
 };
 EXPORT_SYMBOL_GPL(snd_soc_acpi_intel_icl_machines);
 
-static const u64 rt700_0_adr[] = {
-	0x000010025D070000
+static const struct snd_soc_acpi_endpoint single_endpoint = {
+	0,
+	0,
+	0,
+	0,
+};
+
+static const struct snd_soc_acpi_endpoint spk_l_endpoint = {
+	0,
+	1, /* aggregated */
+	0, /* group_position */
+	1, /* group_id */
+};
+
+static const struct snd_soc_acpi_endpoint spk_r_endpoint = {
+	0,
+	1, /* aggregated */
+	1, /* group_position */
+	1, /* group_id */
+};
+
+static const struct snd_soc_acpi_adr_device rt700_0_adr[] = {
+	{
+		0x000010025D070000,
+		1,
+		&single_endpoint,
+	}
 };
 
 static const struct snd_soc_acpi_link_adr icl_rvp[] = {
 	{
 		.mask = BIT(0),
 		.num_adr = ARRAY_SIZE(rt700_0_adr),
-		.adr = rt700_0_adr,
+		.adr_d = rt700_0_adr,
 	},
 	{}
 };
 
-static const u64 rt711_0_adr[] = {
-	0x000010025D071100
+static const struct snd_soc_acpi_adr_device rt711_0_adr[] = {
+	{
+		0x000010025D071100,
+		1,
+		&single_endpoint,
+	}
+
+};
+
+static const struct snd_soc_acpi_adr_device rt1308_1_adr[] = {
+	{
+		0x000110025D130800,
+		1,
+		&single_endpoint,
+	}
 };
 
-static const u64 rt1308_1_adr[] = {
-	0x000110025D130800
+static const struct snd_soc_acpi_adr_device rt1308_2_adr[] = {
+	{
+		0x000210025D130800,
+		1,
+		&single_endpoint,
+	}
 };
 
-static const u64 rt1308_2_adr[] = {
-	0x000210025D130800
+static const struct snd_soc_acpi_adr_device rt1308_1_group1_adr[] = {
+	{
+		0x000110025D130800,
+		1,
+		&spk_l_endpoint,
+	}
 };
 
-static const u64 rt715_3_adr[] = {
-	0x000310025D071500
+static const struct snd_soc_acpi_adr_device rt1308_2_group1_adr[] = {
```

Comment:
looks like rt1308_1_group1_adr[] and rt1308_2_group1_adr[] are never used in soc-acpi-intel-icl-match.c?

Reply:
yes, fixed